feat(MA): adding Marketing Overview

jabahamondes opened the pull request:
## Problem

We weren't able to visualize all the period aggregated information

## Changes

This update enhances the marketing analytics capabilities by allowing for aggregated data queries, improving the overall analytics experience:
- Introduced  to the schema and updated relevant types.
- Added  component to handle aggregated marketing analytics data.
- Implemented utility functions for identifying aggregated queries.
- Updated  to utilize the new  for displaying marketing analytics.
- Enhanced query handling in  to support the new aggregated query type.
- Created  for executing the new query type.
- Refactored existing components to improve code organization and reduce duplication.


## How did you test this code?
Manually, I replaced the real values with strings directly from the ui
<img width="1301" height="503" alt="Screenshot 2025-10-07 at 11 09 52 AM" src="https://github.com/user-attachments/assets/2d04a6c3-1c3d-4bb6-b78d-13d5bd8f6e5a" />

Commit:
feat(marketing): add MarketingAnalyticsAggregatedQuery and related components

- Introduced  to the schema and updated relevant types.
- Added  component to handle aggregated marketing analytics data.
- Implemented utility functions for identifying aggregated queries.
- Updated  to utilize the new  for displaying marketing analytics.
- Enhanced query handling in  to support the new aggregated query type.
- Created  for executing the new query type.
- Refactored existing components to improve code organization and reduce duplication.

This update enhances the marketing analytics capabilities by allowing for aggregated data queries, improving the overall analytics experience.

Commit:
fixing mypy baseline errors

github-actions[bot] committed:
Update E2E screenshots for Playwright

github-actions[bot] committed:
Update E2E screenshots for Playwright

Commit:
Pr's comments

Commit:
Merge branch 'feat/adding-aggregated-marketing-overview' of github.com:PostHog/posthog into feat/adding-aggregated-marketing-overview

jabahamondes commented on products/marketing_analytics/backend/hogql_queries/marketing_analytics_table_query_runner.py:
```diff
@@ -47,94 +39,21 @@ def __init__(self, *args, **kwargs):
             limit_context=self.limit_context, limit=self.query.limit, offset=self.query.offset
         )
-        # Initialize configuration
-        self.config = MarketingAnalyticsConfig()
```
Most of this file was moved into marketing_analytics_base_query_runner.py

Commit:
refactor(marketing): clean up marketing analytics logic and improve query handling

- Removed unnecessary comments in marketing analytics tiles logic.
- Refactored methods in MarketingAnalyticsAggregatedQueryRunner to simplify column handling and improve clarity.
- Updated method names for better understanding of their functionality.
- Enhanced the exclusion logic for specific metrics in column mappings.

These changes streamline the codebase and enhance maintainability while ensuring the functionality remains intact.

Commit:
feat(marketing): add Marketing Analytics Aggregated query type to insights

- Introduced a new query type for Marketing Analytics Aggregated, enhancing the insights capabilities.
- Updated the QUERY_TYPES_METADATA to include the new query type with its name, icon, and menu visibility.

This addition improves the flexibility of analytics queries available in the application.

lricoy committed:
Merge branch 'master' into feat/adding-aggregated-marketing-overview

Commit:
fixing tests

Commit:
Merge branch 'feat/adding-aggregated-marketing-overview' of github.com:PostHog/posthog into feat/adding-aggregated-marketing-overview

lricoy reviewed (approved):
Please let me know when you deploy this so we can keep an eye for it. I don't expect any big problem but since it includes a refactor outside Marketing Analytics, it is better safer than sorry.

Great job!

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-aggregated-marketing-overview

Commit:
feat(overview): add compact mode for OverviewGrid and update EvenlyDistributedRows

- Introduced compact mode in OverviewGrid for a more space-efficient layout.
- Updated min-width constants and CSS classes for compact and default views.
- Enhanced OverviewItemCellSkeleton to adapt to compact mode.
- Modified EvenlyDistributedRows to accept maxItemsPerRow prop for better item distribution.
- Updated MarketingAnalyticsOverview to utilize the new compact mode.

Commit:
fixing test

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-aggregated-marketing-overview

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

Commit:
reverting gap

Commit:
Merge branch 'feat/adding-aggregated-marketing-overview' of github.com:PostHog/posthog into feat/adding-aggregated-marketing-overview

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-aggregated-marketing-overview

Commit:
Merge branch 'feat/adding-aggregated-marketing-overview' of github.com:PostHog/posthog into feat/adding-aggregated-marketing-overview
